laterpay/application/Controller/Admin/Account.php

```diff
@@ -199,24 +216,33 @@ protected static function update_api_key( LaterPay_Core_Event $event ) {
         if ( $event->has_argument( 'is_live' ) ) {
             $is_live = $event->get_argument( 'is_live' );
         }
-        $api_key_form       = new LaterPay_Form_ApiKey( $_POST );
-        $api_key            = $api_key_form->get_field_value( 'api_key' );
-        $api_key_type       = $is_live ? 'live' : 'sandbox';
-        $transaction_type   = $is_live ? 'REAL' : 'TEST';
-
-        if ( $api_key_form->is_valid() ) {
-            update_option( sprintf( 'laterpay_%s_api_key', $api_key_type ), $api_key );
+        $api_key_form     = new LaterPay_Form_ApiKey( $_POST );
+        $api_key          = $api_key_form->get_field_value( 'api_key' );
+        $api_key_type     = $is_live ? 'live' : 'sandbox';
+        $transaction_type = $is_live ? 'REAL' : 'TEST';
+
+        try {
+            $api_key_form->validate( $_POST );
+        } catch ( LaterPay_Core_Exception_FormValidation $e ) {
```

Comment:
need to discuss this case, generating exception from the form is not good, nothing illegal should happen while this.
its more better to throw exception up, if form is not valid.
one more thing, event should be stopped